parsers/ocaml.c

```diff
@@ -1509,38 +1632,57 @@ static void moduleSpecif (vString * const ident, ocaToken what)
 	case OcaKEYWORD_sig:
 		pushStrongContext (lastModule, ContextModule);
 		toDoNext = &globalScope;
+		needStrongPoping = FALSE;
 		break;
 
 	case Tok_PARL:	/* ( */
 		toDoNext = &contextualTillToken;
 		comeAfter = &globalScope;
 		waitedToken = Tok_PARR;
-		contextualTillToken (ident, what);
+		contextualTillToken (ident, what, whatNext);
+		break;
+
+	case Tok_Of:
+	case Tok_EQ:
 		break;
 
 	default:
 		vStringClear (lastModule);
 		toDoNext = &globalScope;
+		break;
 	}
 }
 
 /* parse :
  * module name = ...
  * then pass the token stream to moduleSpecif */
-static void moduleDecl (vString * const ident, ocaToken what)
+
```

Comment:
I don't think adding an empty line between the comment and function is a good idea